ctparse/nb_scorer.py

```diff
@@ -64,7 +64,10 @@ def score_final(
         # NOTE: the prediction is log-odds, or logit
         model_score = pred[0][1] - pred[0][0]
 
-        return model_score + len_score
+        # We want the len_score to always take precedence. I believe a logit won't go up more
```

Comment:
As long as we can come up with a way to translate this into a scalar for the `CTParse` the internal changes would be rather small.
How come we "know" that we should first order by length and then the model?

Reply:
Simple Answer: One of the reason is that with this length scaling we get rid of an important regression while still maintaining performance. This comes up in simple expressions

Complicated answer: I believe to be the reason for this effect

```
15.10.2020 (ref date 14 oct 2020)
15.10 (ref date 14 oct 2020)
```

Expression 1 will match both the rule `15 October + latent_year` and `15 October + 2020`, while expression 2 will match only `15 October + latent_year`

Since in our dataset we mostly have that the reference date is same year as the one in the time expression, what happens is that the probability of the sequence that end in latent year gets a larger probability score than the one with the year, and the `len_score` is too small to deal with it (a half match would have a len score log(0.5) versus a full match log(1) which is only 0.3, logits for those highly imbalanced sequences can go 15-16).

Why would it appear in this particular PR? I believe it is because this affects the number of "negatives" since we remove certain productions for the strings of type `15.10`